ArcGISRuntimeSDKQt_CppSamples/Maps/CreateDynamicBasemapGallery/CreateDynamicBasemapGallery.h

```diff
@@ -0,0 +1,94 @@
+// [WriteFile Name=CreateDynamicBasemapGallery, Category=Maps]
+// [Legal]
+// Copyright 2024 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifndef CREATEDYNAMICBASEMAPGALLERY_H
+#define CREATEDYNAMICBASEMAPGALLERY_H
+
+#include <QAbstractListModel>
+#include <QMap>
+#include <QObject>
+#include <QUrl>
+
+#include "BasemapStyleInfo.h"
+#include "Viewpoint.h"
+
+Q_MOC_INCLUDE("MapQuickView.h")
+Q_MOC_INCLUDE("BasemapStyleInfo.h")
+Q_MOC_INCLUDE("BasemapStyleListModel.h")
+
+class BasemapStyleListModel;
+
+namespace Esri::ArcGISRuntime {
+enum class BasemapStyleLanguageStrategy;
+class Map;
+class MapQuickView;
+class Viewpoint;
+} // namespace Esri::ArcGISRuntime
+
+class CreateDynamicBasemapGallery : public QObject
+{
+    Q_OBJECT
+
+    Q_PROPERTY(Esri::ArcGISRuntime::MapQuickView *mapView READ mapView WRITE setMapView NOTIFY
+                   mapViewChanged)
+    Q_PROPERTY(const BasemapStyleListModel* const gallery READ gallery NOTIFY galleryChanged)
+    Q_PROPERTY(const QList<QString>& languageStrategies READ languageStrategies NOTIFY languageStrategiesChanged)
+    Q_PROPERTY(const QList<QString>& languages READ languages NOTIFY languagesChanged)
```

Comment:
`QStringList` used instead of `QList<QString>` in https://github.com/Esri/arcgis-maps-sdk-samples-qt/pull/1714/commits/d975f84a03f5b81baaeaadb38bd347bbf293e61b.